client/app/bundles/shared/components/Dropdown/DropdownFactory.jsx

```diff
@@ -0,0 +1,36 @@
+// @flow
+import React from 'react';
+import css from './Dropdown.scss';
+
+type Option = {
+  key?: string,
+  label: string,
+  value: any,
+};
+
+type Props = {
+  onChange?: (event: Event) => any,
+  options: Option[],
+  value?: any,
+};
+
+const DropdownFactory = (variationClassName: string) => {
+  const DropdownComponent = ({ onChange, options, value: propValue }: Props) => (
+    <select className={`${css.dropdown} ${variationClassName}`} onChange={onChange} value={propValue}>
+      {options.map(({ key, label, value }: Option) => (
+        <option key={key || (value !== null && value !== undefined ? value : label)} value={value}>
+          {label}
+        </option>
+      ))}
+    </select>
+  );
+  DropdownComponent.defaultProps = {
+    onChange: () => {},
+    options: [],
+    value: undefined,
```

Comment:
I think you can just exclude line 30 and value will be implicitly undefined.

Reply:
I'll get back to you on this... I think I had to include it for lints to pass. But I'll double check.